test/cli/compiler/test-migrations.js

```diff
@@ -1,35 +1,50 @@
 const test = require("tape"); // https://github.com/substack/tape
+const colorize = require('tap-colorize');
 const path = require("path");
-const process = require("process");
 const testUtils = require("../../../bin/tools/utils");
 const fsp = require("fs").promises;
 const digest = require("util").promisify(require("dirsum").digest);
-//const qx = path.resolve(__dirname, "..", "..", "..", "bin", "source", "qx");
-const qx = testUtils.getCompiler();
-const debug = false;
+const qxCmdPath = testUtils.getCompiler();
 
-const debugArg = debug ? "--debug" : "";
-test("Test migrations", async t => {
-  try {
-    const baseDir = path.join(__dirname, "test-migrations", "v6.0.0");
-    const unmigratedDir = path.join(baseDir, "unmigrated");
-    const migratedDir = path.join(baseDir, "migrated");
-    await testUtils.deleteRecursive(migratedDir);
-    await testUtils.sync(unmigratedDir, migratedDir);
-    t.comment("Upgrade alert");
-    let result = await testUtils.runCommand(migratedDir, qx, "clean", debugArg);
-    t.match(result.error, /There are 6 new migrations/);
-    t.comment("Dry run");
-    result = await testUtils.runCommand(migratedDir, qx, "migrate", "--verbose", "--dry-run", "--max-version=6.0.0", debugArg);
-    t.match(result.output, /0 migrations applied, 7 migrations pending/);
-    t.comment("Run migration");
-    result = await testUtils.runCommand(migratedDir, qx, "migrate", "--verbose", "--max-version=6.0.0", debugArg);
-    t.match(result.output, /11 migrations applied, 0 migrations pending/);
-    let checksum = (await digest(migratedDir,'sha1')).hash;
-    t.comment(`Checksum of migrated app: ${checksum}`);
-    t.equals(checksum, "8c1c8dc402ae7eb3f72941284b79bf4cf34c90fd");
-    t.end();
-  } catch(ex) {
-    t.end(ex);
+// colorize output
+test.createStream().pipe(colorize()).pipe(process.stdout);
+
+// debugging
+const debug = true;
+const debugArg = debug ? "--debug --colorize" : "--colorize";
+
+function testMigration(maxVersion, numMigrationsExpected, checksumExpected) {
+  return async t => {
+    try {
+      t.comment(`>>>>>>>>>>>>>>>>>> TESTING v${maxVersion} migration <<<<<<<<<<<<<<<<<<<`);
```

Comment:
what is `t` here?  looks like it's `tape`

Reply:
exactly - that's why it is not good to name the object `assert`

Reply:
and the function returns a function that can be fed into the tape test function, that's why it looks like it.

Reply:
renaming it to `tape` would be better than `t`, because `t.comment` looks like `this.comment` ... the reason I asked what `t` is, is because I had to go through almost the entire file to follow what was happening, and that costs time.  I could  guess that `t` was probably not `this`, but I couldn't tell if it was the outer scope `t` either.